src/ad/abstract.js

```diff
@@ -0,0 +1,152 @@
+import { SortedListItem } from '../util/sorted-list-item'
+import { SortedList } from '../util/sorted-list'
+
+/**
+ * Abstract class for ads (i.e., {@link InLine} and {@link Wrapper}).
+ * @class AbstractAd
+ * @abstract
+ * @protected
+ * @extends SortedListItem
+ * @author Tim De Pauw <tim.depauw@zentrick.com>
+ * @copyright © 2018 Zentrick nv
+ */
+export class AbstractAd extends SortedListItem {
+  constructor () {
+    super()
+    this._impressions = []
```

Comment:
This is an example of a non-nullable array, which works better for me than the current implementation of `Creatives`. I've added a comment there to make it consistent.